runtime/pkg/metricssql/filter_parser_test.go

```diff
@@ -0,0 +1,116 @@
+package metricssqlparser_test
+
+import (
+	reflect "reflect"
+	"testing"
+
+	"github.com/rilldata/rill/runtime/metricsview"
+	metricssqlparser "github.com/rilldata/rill/runtime/pkg/metricssql"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseSQLFilter(t *testing.T) {
+	tests := []struct {
+		name    string
+		sql     string
+		want    *metricsview.Expression
+		wantErr bool
+	}{
+		{
+			"equal expression",
+			"dim = 'val'",
+			&metricsview.Expression{
+				Condition: &metricsview.Condition{
+					Operator: metricsview.OperatorEq,
+					Expressions: []*metricsview.Expression{
+						{
+							Name: "dim",
+						},
+						{
+							Value: "val",
+						},
+					},
+				},
+			},
+			false,
+		},
+		{
+			"in expression",
+			"dim IN ('helllo', 'world')",
+			&metricsview.Expression{
+				Condition: &metricsview.Condition{
+					Operator: metricsview.OperatorIn,
+					Expressions: []*metricsview.Expression{
+						{
+							Name: "dim",
+						},
+						{
+							Value: "helllo",
+						},
+						{
+							Value: "world",
+						},
```

Comment:
Since `IN` is a binary operator, can we change this to output two sub-expressions where the second one is a slice of values instead? I.e. `[{"name": "dim"}, {"value": ["helllo", "world"]}]`? 

I know the flattened approach is also supported (because the UI does it), but that's just for backwards compatibility (and I hope we can eventually deprecate it entirely): https://github.com/rilldata/rill/blob/d66a28ff94b55273d102cc2112faf8d05465d254/runtime/metricsview/astexpr.go#L157-L175